python/google/protobuf/internal/well_known_types.py

```diff
@@ -750,6 +750,23 @@ def _GetStructValue(struct_value):
     raise ValueError('Value not set')
 
 
+def _StructToDict(struct_value):
```

Comment:
Current implementation is a little confusing because struct_value can be a List or other type. From the _StructToDict() name, only Struct should call this function. The implement can be this:
```
def _StructToDict(struct_value):
    result = {}
    for key, value in struct_value.items():
       if isinstance(value, Struct):
            ...
       elif isinstance(value, ListValue):
            ...  # may need to add the to_list method
       elif getattr(struct_value, 'WhichOneof', None):
            ...
       else:
            ...
    return result
```